hazelcast/include/hazelcast/client/serialization/generic_record_builder.h

```diff
@@ -0,0 +1,980 @@
+     * @see set_timestamp()
+     */
+    generic_record_builder& set_array_of_timestamp(
+      std::string field,
+      boost::optional<std::vector<boost::optional<local_date_time>>> value);
+
+    /**
+     * It is legal to set the field again only when Builder is created with
+     * {@link generic_record::new_builder_with_clone()}. Otherwise, it is
+     * illegal to set to the same field twice. Array items can not be null.
+     *
+     * @param fieldName name of the field as it is defined in its schema
+     * definition.
+     * @param value     to set to generic_record
+     * @return itself for chaining
+     * @throws exception::hazelcast_serialization if the field name does not
+     * exist in the schema definition or the type of the field does not match
+     * the one in the schema definition or the same field is trying to be set
+     * without using
+     *                                            {@link
+     * generic_record::new_builder_with_clone()}.
+     * @see set_timestamp_with_timezone()
+     */
+    generic_record_builder& set_array_of_timestamp_with_timezone(
+      std::string field,
+      boost::optional<std::vector<boost::optional<offset_date_time>>> value);
+
+    /**
+     * It is legal to set the field again only when Builder is created with
+     * {@link generic_record::new_builder_with_clone()}. Otherwise, it is
+     * illegal to set to the same field twice. This method allows nested
+     * structures. Subclasses should also be created as `GenericRecord`. <p>
+     * Array items can not be null.
+     * <p>
+     * For Compact objects,
+     * it is not allowed write an array containing different item
+     * types or a {@link exception::hazelcast_serialization} will be thrown.
+     *
+     * @param fieldName name of the field as it is defined in its schema
+     * definition.
+     * @param value     to set to generic_record
+     * @return itself for chaining
+     * @throws exception::hazelcast_serialization if the field name does not
+     * exist in the schema definition or the type of the field does not match
+     * the one in the schema definition or the same field is trying to be set
+     * without using
+     *                                            {@link
+     * generic_record::new_builder_with_clone()}.
+     */
+    generic_record_builder& set_array_of_generic_record(
+      std::string field,
+      boost::optional<std::vector<boost::optional<generic_record>>> value);
+
+private:
+    enum class strategy
+    {
+        normal,
+        cloner,
+        schema_bounded
+    };
+
+    friend class generic_record;
+    generic_record_builder(
+      pimpl::schema); // DeserializedSchemaBoundGenericRecordBuilder
+    generic_record_builder(
+      pimpl::schema,
+      std::unordered_map<std::string,
+                         boost::any>); // DeserializedGenericRecordCloner
+
+    template<typename T>
+    generic_record_builder& write(std::string field_name,
```

Comment:
In Java side, it seems there are three classes, in C++ we have ane class for builder. And that makes this method longer than it has to be. Might we split this method at least for each strategy.

Reply:
Method is a bit complex but I think it is understandable. Also those classes have common behaviours so those functions will be partially duplicated in that case.